lib/agent.js

```diff
@@ -459,7 +459,9 @@ Agent.prototype.flush = function (cb) {
     // TODO: Only bind the callback if the transport can't use AsyncResource from async hooks
     this._transport.flush(cb && this._instrumentation.bindFunction(cb))
   } else {
-    this.logger.warn(new Error('cannot flush agent before it is started'))
+    // Log an *err* to provide a stack for the user.
+    const err = new Error('cannot flush agent before it is started')
+    this.logger.warn({ err }, err.message)
```

Comment:
This is interesting to me -- I didn't know about pino's optional `mergingObject` parameter/argument until [researching this syntax](https://getpino.io/#/docs/api?id=trace). (nothing to see/do here, carry on)

Reply:
It originates from https://github.com/trentm/node-bunyan#log-method-api but there are some minor differences.
- pino dropped the `log.info()`-with-no-args call signature for the more traditional log4j `log.isLevelEnabled(level)` (https://getpino.io/#/docs/api?id=islevelenabled) which was a good call.
- Bunyan treats `log.info(err)` and `log.info({err: err})` the same. Pino does not. The form that logs the error with the `err` field name is used by `@elastic/ecs-pino-format` module to format the error fields into ECS format, so we should bias to using that form.
- Bunyan will use `err.message` for the "message" field with `log.info({err: err})`. Pino will leave the message empty. Arguably pino's is cleaner, but for the (rare) case of an APM Agent user specifying a custom `logger`, they will only get the "message" field, so it is nice to have *some* message there.